apache2/ag_mdb/ag_mdb.cpp

```diff
@@ -57,36 +56,91 @@ int Lock_create(struct agmdb_lock *new_lock, const char* lock_name, int lock_nam
-    new_mutex = CreateMutex(
-                    NULL,               // Default security settings.
-                    FALSE,              // Do not take the lock after created.
-                    read_lock_name);    // The name of read lock.
-    if(new_mutex == NULL)
+    new_lock->read_lock_handle = CreateMutex(
+                                    NULL,               // Default security settings.
+                                    FALSE,              // Do not take the lock after created.
+                                    read_lock_name);    // The name of read lock.
+    if(new_lock->read_lock_handle == NULL)
         return AGMDB_ERROR_LOCK_WIN_MUTEX_CREATE_FAIL;
     if(GetLastError() == ERROR_ALREADY_EXISTS)
         lock_exists = true;
-    new_lock->read_lock_handle = OpenMutex(
-                                    MUTEX_ALL_ACCESS,   //
-                                    false,              //
-                                    read_lock_name);
     
-    new_mutex = CreateMutex(
-                    NULL,               // Default security settings.
-                    FALSE,              // Do not take the lock after created.
-                    write_lock_name);   // The name of write lock.
-    if(new_mutex == NULL)
+    new_lock->write_lock_handle = CreateMutex(
+                                    NULL,               // Default security settings.
+                                    FALSE,              // Do not take the lock after created.
+                                    write_lock_name);   // The name of write lock.
+    if(new_lock->write_lock_handle == NULL) {
+        CloseHandle(new_lock->read_lock_handle);
+        new_lock->read_lock_handle = NULL;
         return AGMDB_ERROR_LOCK_WIN_MUTEX_CREATE_FAIL;
-    if((GetLastError() == ERROR_ALREADY_EXISTS) != lock_exists) // One lock exists, another not.
+    }
+        
+    if((GetLastError() == ERROR_ALREADY_EXISTS) != lock_exists) {
+        // One lock exists, another not.
+        CloseHandle(new_lock->read_lock_handle);
+        new_lock->read_lock_handle = NULL;
+        CloseHandle(new_lock->write_lock_handle);
+        new_lock->write_lock_handle = NULL;
         return AGMDB_ERROR_LOCK_WIN_ONLY_ONE_LOCK_EXISTS;
-    new_lock->write_lock_handle = OpenMutex(
-                                    MUTEX_ALL_ACCESS,   //
-                                    false,              //
-                                    write_lock_name);
+    }
+
+    if(lock_exists == true)
+        return AGMDB_SUCCESS_LOCK_OPEN;
+    else
+        return AGMDB_SUCCESS_LOCK_CREATE;
 #endif
     // Should never get here
     return AGMDB_SUCCESS_LOCK_CREATE;
 }
 
+/**
+ ** Destroy the lock of AG Memory Database 
+ ** @param db_lock: The agmdb_lock sturcture  
+ ** return: AGMDB_SUCCESS if successfully destroy the lock, 
+ **         AGMDB_ERROR if failed.
+ */
+int Lock_destroy(struct agmdb_lock *db_lock) {
+    /* Windows doesn't do anything */
```

Comment:
so in windows, if the process exit, the lock will be automatically destroyed by the OS?